generator/vulkan/render.zig

```diff
@@ -1035,13 +1035,80 @@ fn Renderer(comptime WriterType: type) type {
                 \\    const Info = struct {
                 \\        name: [:0]const u8,
                 \\        version: u32,
+                \\        instance_functions: InstanceCommandFlags,
+                \\        device_functions: DeviceCommandFlags,
                 \\    };
             );
+            // The commands in an extension are not pre-sorted based on if they are instance or device functions.
+            var instance_commands = std.ArrayList([]const u8).init(self.allocator);
+            defer instance_commands.deinit();
+            var device_commands = std.ArrayList([]const u8).init(self.allocator);
+            defer device_commands.deinit();
             for (self.registry.extensions) |ext| {
                 try self.writer.writeAll("pub const ");
                 try self.writeIdentifierWithCase(.snake, trimVkNamespace(ext.name));
                 try self.writer.writeAll("= Info {\n");
                 try self.writer.print(".name = \"{s}\", .version = {},", .{ ext.name, ext.version });
+                // collect extension functions
+                for(ext.requires) |require| {
+                    for(require.commands) |command_name| {
+                        const decl = self.resolveDeclaration(command_name) orelse continue;
+                        // If the target type does not exist, it was likely an empty enum -
+                        // assume spec is correct and that this was not a function alias.
+                        const decl_type = self.resolveAlias(decl) catch continue;
+                        const command = switch (decl_type) {
+                            .command => |cmd| cmd,
+                            else => continue,
+                        };
+                        const class = classifyCommandDispatch(command_name, command);
+                        switch (class) {
+                            // Vulkan extensions cannot add base functions.
+                            .base => continue,
+                            // TODO: some extensions have multiple separate requirements.
+                            // For example, "VK_KHR_push_descriptor" depends on either "VK_VERSION_1_1" or "VK_KHR_descriptor_update_template",
+                            // which causes the same function to be mentioned twice; once for each requirement.
+                            // For now, duplicate functions are just removed,
```

Comment:
I feel like theres not much of a problem to add these functions multiple times. Usually one will OR them together using `merge()`, so there is no problem, and it feels more natural anyway. Besides, the logic is currently quadratic. I suggest to just get rid of that code and allow multiple defintions.